consensus/src/checkpoint.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::{Chain, ProtocolStateChain};
+use mina_crypto::base58::Base58Encodable;
+use mina_crypto::hash::{EpochSeed, StateHash};
+use mina_rs_base::protocol_state::ProtocolState;
+
+pub const SLOTS_PER_EPOCH: u32 = 7140;
+
+#[derive(Debug)]
+pub enum ConsensusErrTyp {
+    ConsensusInitFail,
+}
+pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusErrTyp> {
+    genesis.body.consensus_state.staking_epoch_data.seed = EpochSeed::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .start_checkpoint = StateHash::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .lock_checkpoint = StateHash::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .epoch_length
+        .0 = 1;
+    genesis.body.consensus_state.next_epoch_data.seed =
+        Base58Encodable::from_base58("2vaRh7FQ5wSzmpFReF9gcRKjv48CcJvHs25aqb3SSZiPgHQBy5Dt")
+            .map_err(|_| ConsensusErrTyp::ConsensusInitFail)?;
+    genesis
+        .body
+        .consensus_state
+        .next_epoch_data
+        .start_checkpoint = StateHash::default();
+    genesis.body.consensus_state.next_epoch_data.lock_checkpoint =
+        Base58Encodable::from_base58("3NLoKn22eMnyQ7rxh5pxB6vBA3XhSAhhrf7akdqS6HbAKD14Dh1d")
+            .map_err(|_| ConsensusErrTyp::ConsensusInitFail)?;
+    genesis.body.consensus_state.next_epoch_data.epoch_length.0 = 2;
+    Ok(())
+}
+
+pub fn is_short_range(
+    c0: &ProtocolStateChain,
+    c1: &ProtocolStateChain,
+) -> Result<bool, ConsensusErrTyp> {
```

Comment:
OK for now, but if u look at [ocaml code](https://github.com/MinaProtocol/mina/blob/9dd0c963e87f80c81006c6ecda34efbf6543d2a3/src/lib/consensus/proof_of_stake.ml#L2919), I think we can safely return false for the err condition

Reply:
For not use unwrap() https://github.com/ChainSafe/mina-rs/pull/80/commits/c15385f425f4344052ef57f3e27bcee1a5f621c2#